code/game/gamemodes/dynamic/dynamic_rulesets.dm

```diff
@@ -201,6 +218,41 @@
 /// Only called if ruleset is flagged as HIGH_IMPACT_RULESET
 /datum/dynamic_ruleset/proc/round_result()
 
+
+/// Checks if the ruleset is "dead", where all the antags are either dead or deconverted.
+/datum/dynamic_ruleset/proc/is_dead()
+	. = TRUE
+	for(var/datum/mind/mind in assigned)
+		var/mob/living/body = mind.current
+		// If they have no body, they're dead for realsies.
+		if(QDELETED(body))
+			continue
+		// Well, if there's nobody in the body, they might as well be dead.
+		if(!body.ckey)
+			continue
+		// Have they been AFK for over 20 minutes? If so, eh, we won't take them into consdideration.
+		if(body.client?.is_afk(20 MINUTES))
+			continue
+		// Have they been husked by a non-burn source? Probably really really dead.
+		if(HAS_TRAIT_NOT_FROM(body, TRAIT_HUSK, "burn"))
+			continue
+		// Alright, they have a body with a ckey, but are they actually dead?
+		if(body.stat == DEAD)
+			// Has their soul departed or been ripped out? If so, yep, they dead alright.
+			if(body.soul_departed() || mind.hellbound)
+				continue
+			// Are they in medbay or an operating table/stasis bed, and have been dead for less than 20 minutes? If so, they're probably being revived.
+			if(world.time <= (mind.last_death + 20 MINUTES) && (istype(get_area(body), /area/medical) || (locate(/obj/machinery/stasis) in body.loc) || (locate(/obj/structure/table/optable) in body.loc)))
```

Comment:
I would probably lower this to 15 minutes, as the death time is slowed by stasis beds, if you're using that type of death time..

Maybe both, so 20 minutes since death OR body is decayed
Generally when a body gets to the point of decaying it will be a long while before they are revived if they're not in cloning

Reply:
Changed to 15 minutes